resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -61,10 +61,8 @@ private[spark] class KubernetesClusterSchedulerBackend(
   private val runningExecutorsToPods = new mutable.HashMap[String, Pod]
   // Indexed by executor pod names and guarded by RUNNING_EXECUTOR_PODS_LOCK.
   private val runningPodsToExecutors = new mutable.HashMap[String, String]
-  // TODO(varun): Get rid of this lock object by my making the underlying map a concurrent hash map.
-  private val EXECUTOR_PODS_BY_IPS_LOCK = new Object
-  // Indexed by executor IP addrs and guarded by EXECUTOR_PODS_BY_IPS_LOCK
-  private val executorPodsByIPs = new mutable.HashMap[String, Pod]
+  private val executorPodsByIPs: concurrent.Map[String, Pod] = new
```

Comment:
I've looked into databrick scala recomendations, they suggest to avoid concurrent.Map and use j.u.c.ConcurrentHashMap.

> Prefer java.util.concurrent.ConcurrentHashMap over scala.collection.concurrent.Map
> https://github.com/databricks/scala-style-guide#concurrency